### PR TITLE
New addrbook lua for migration

### PR DESCRIPTION
**Pull request by kadavr**

### CODE REVIEW

File: addrbook.lua [hunk 1]
```diff
@@ -23,3 +23,41 @@ function addrbook_add_recipient(user_id, rcp_email, rcp_name, timestamp)
 	box.update(1, { user_id, rcp_email }, "+p", 4, 1)
 	return 0 -- contact weight updated only
 end
+
+function addrbook_load(user_id)
+	user_id = box.unpack('i', user_id)
+	if(uid == nil) then
```
Comment (derElektrobesen) [inline addrbook.lua]: 1. uid --> user_id
2. codestyle
3. check user_id before unpack

File: addrbook.lua [hunk 2]
```diff
@@ -23,3 +23,41 @@ function addrbook_add_recipient(user_id, rcp_email, rcp_name, timestamp)
...
+		return 0
+	end
+	local tuple = box.select(0, 0, user_id) -- space 0 index 0
```
Comment (derElektrobesen) [inline addrbook.lua]: Comment what is space 0 (and index 0) on the file head

File: addrbook.lua [hunk 3]
```diff
@@ -23,3 +23,41 @@ function addrbook_add_recipient(user_id, rcp_email, rcp_name, timestamp)
...
+	if( tuple == nil) then
```
Comment (derElektrobesen) [inline addrbook.lua]: 1. codestyle
2. You trying to return a tuple or 0 O_o
    Return tuple or nil. It should be ok.

File: addrbook.lua [hunk 4]
```diff
@@ -23,3 +23,41 @@ function addrbook_add_recipient(user_id, rcp_email, rcp_name, timestamp)
...
+		return 0
+	end
+	return tuple
+end
+
+function addrbook_save(user_id, book)
+	user_id = box.unpack('i', user_id)
+	if(user_id == nil or book == nil) then
```
Comment (derElektrobesen) [inline addrbook.lua]: Check user_id before unpack

File: addrbook.lua [hunk 5]
```diff
@@ -23,3 +23,41 @@ function addrbook_add_recipient(user_id, rcp_email, rcp_name, timestamp)
...
+		return 0
+	end
+	local tuple = box.update(0, user_id, "=p", 1, book)
```
Comment (derElektrobesen) [inline addrbook.lua]: Same as above

File: addrbook.lua [hunk 6]
```diff
@@ -23,3 +23,41 @@ function addrbook_add_recipient(user_id, rcp_email, rcp_name, timestamp)
...
+
+	if( tuple == nil) then
+		return 0
+	end
+	return tuple
+end
+
+function addrbook_delete(user_id)
```
Comment (derElektrobesen) [inline addrbook.lua]: check user_id before unpack, return nil.
Codestyle.

File: addrbook.lua [hunk 7]
```diff
@@ -23,3 +27,30 @@ function addrbook_add_recipient(user_id, rcp_email, rcp_name, timestamp)
 	box.update(1, { user_id, rcp_email }, "+p", 4, 1)
 	return 0 -- contact weight updated only
 end
+
+function addrbook_load( user_id )
+	if( user_id == nil ) then
```
Comment (derElektrobesen) [inline addrbook.lua]: Syntax error.
If statement should be used without brackets.

File: addrbook.lua [hunk 8]
```diff
@@ -23,3 +27,30 @@ function addrbook_add_recipient(user_id, rcp_email, rcp_name, timestamp)
...
+		return nil
+	end
+	user_id = box.unpack( 'i', user_id )
```
Comment (derElektrobesen) [inline addrbook.lua]: Just write " return box.select(0,0, user_id)"
And remove spaces after brackets.

Comment is absolutely worthless.

**Comment by jirfag** [comment]
Понадобился еще один фикс, чтобы работал insert, сейчас работает только udpate

**Pull request closed without merging**